### PR TITLE
Added nRFCrypto_AES Class for AES encryption/decryption

### PR DESCRIPTION
**Pull request by Kongduino**
Since the main goal of a "crypto" library is to have encryption, and the CC310 library has AES included, I thought we should take advantage of that and provide at least a few options for this. This first release supplies AES ECB and CBC through a `nRFCrypto_AES` class.

```c
class nRFCrypto_AES {
  public:
    nRFCrypto_AES(void);
    bool begin(void);
    void end(void);
    int Process(char *msg, uint8_t msgLen, uint8_t *IV, uint8_t *pKey, uint8_t pKeyLen,
                   char *retBuf, SaSiAesEncryptMode_t modeFlag, SaSiAesOperationMode_t opMode);
    SaSiAesEncryptMode_t encryptFlag = (SaSiAesEncryptMode_t) 0; // SASI_AES_ENCRYPT
    SaSiAesEncryptMode_t decryptFlag = (SaSiAesEncryptMode_t) 1; // SASI_AES_DECRYPT
    SaSiAesOperationMode_t ecbMode = (SaSiAesOperationMode_t) 0; // SASI_AES_MODE_ECB
    SaSiAesOperationMode_t cbcMode = (SaSiAesOperationMode_t) 1; // SASI_AES_MODE_CBC
  private:
    bool _begun;
    SaSiAesPaddingType_t _paddingType = (SaSiAesPaddingType_t) 0; // SASI_AES_PADDING_NONE
    SaSiAesKeyType_t _userKey = (SaSiAesKeyType_t) 0; // SASI_AES_USER_KEY
};
```

The `nRFCrypto_AES::Process` function does everything:

* ECB/CBC
* encrypt/decrypt.

The sample code [here](https://github.com/Kongduino/RAK4631_AES) shows how to do all 4 operations.

### CODE REVIEW

File: src/Adafruit_nRFCrypto.h [hunk 1]
```diff
@@ -1,58 +1,50 @@
...
+   The above copyright notice and this permission notice shall be included in
+   all copies or substantial portions of the Software.
+   THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+   IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+   FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+   AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+   LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+   OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+   THE SOFTWARE.
+*/
```
Comment (hathach) [inline src/Adafruit_nRFCrypto.h:19]: please revert this license changes

File: src/Adafruit_nRFCrypto.cpp [hunk 2]
```diff
@@ -95,117 +90,114 @@ void Adafruit_nRFCrypto::end(void)
...
-  { .key =  CRYS_ECIES_INVALID_KDF_DERIV_MODE_ERROR                      , .data = "ECIES_INVALID_KDF_DERIV_MODE" },
-  { .key =  CRYS_ECIES_INVALID_KDF_HASH_MODE_ERROR                       , .data = "ECIES_INVALID_KDF_HASH_MODE" },
-  { .key =  CRYS_ECIES_INVALID_SECRET_KEY_PTR_ERROR                      , .data = "ECIES_INVALID_SECRET_KEY_PTR" },
-  { .key =  CRYS_ECIES_INVALID_SECRET_KEY_SIZE_ERROR                     , .data = "ECIES_INVALID_SECRET_KEY_SIZE" },
-  { .key =  CRYS_ECIES_INVALID_CIPHER_DATA_PTR_ERROR                     , .data = "ECIES_INVALID_CIPHER_DATA_PTR" },
-  { .key =  CRYS_ECIES_INVALID_CIPHER_DATA_SIZE_PTR_ERROR                , .data = "ECIES_INVALID_CIPHER_DATA_SIZE_PTR" },
-  { .key =  CRYS_ECIES_INVALID_CIPHER_DATA_SIZE_ERROR                    , .data = "ECIES_INVALID_CIPHER_DATA_SIZE" },
-  { .key =  CRYS_ECIES_INVALID_TEMP_DATA_PTR_ERROR                       , .data = "ECIES_INVALID_TEMP_DATA_PTR" },
-  { .key =  CRYS_ECIES_INVALID_EPHEM_KEY_PAIR_PTR_ERROR                  , .data = "ECIES_INVALID_EPHEM_KEY_PAIR_PTR" }
+  { .key =  CRYS_ECPKI_ILLEGAL_DOMAIN_ID_ERROR, .data = "ECPKI_ILLEGAL_DOMAIN_ID" },
```
Comment (hathach) [inline src/Adafruit_nRFCrypto.cpp:93]: would you mind explaining why making this change

**Review by hathach — COMMENTED** [review]
Thank you for your PR, would you mind 
- adding simple AES example similar to other example to demonstrate how to use the API.
- tell me more why would you make some white space/indentation changes, since some of them does not improve readibility.

PS: Would you also update/merge this PR with master branch which is just added ci build. Since I notice in your previous commit, it removed the  `Adafruit_TinyUSB include` along with guard for `NRF_CRYPTOCELL` which is absolutely required for building with https://github.com/adafruit/Adafruit_nRF52_Arduino

**Comment by Kongduino** [comment]
I had reformatted because we probably have different views about what is readable and what isn't... I will close this PR and start anew.

**Comment by hathach** [comment]
> I had reformatted because we probably have different views about what is readable and what isn't... I will close this PR and start anew.

you don't have to make an new PR, just push more update to the existing PR. The less work the better. Though it is up to you.

**Pull request closed without merging**